cmd/thanos/compact.go

```diff
@@ -786,6 +789,9 @@ func (cc *compactConfig) registerFlag(cmd extkingpin.FlagClause) {
 		"NOTE: This flag is ignored and (enabled) when --deduplication.replica-label flag is set.").
 		Hidden().Default("false").BoolVar(&cc.enableVerticalCompaction)
 
+	cmd.Flag("compact.remove-overlapping", "In house flag to remove overlapping blocks. Turn this on to fix PLAT-104290.").
```

Comment:
No point in putting a Jira ticket number in a public repo here. Just briefly describe the issue.

Reply:
aha, forgot this is public, let me fix it.